File: Java/ex2.java
class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    public String detalhes() {
        return "Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano;
    }

    public String acelerar(int incremento) {
        velocidade += incremento;
        return marca + " " + modelo + " acelerou. Velocidade atual: " + velocidade + " km/h";
    }

    public String frear(int decremento) {
        velocidade = Math.max(0, velocidade - decremento);
        return marca + " " + modelo + " freou. Velocidade atual: " + velocidade + " km/h";
    }

    public String exibirVelocidade() {
        return "Velocidade atual do " + marca + " " + modelo + ": " + velocidade + " km/h";
    }
}

public class Main {
    public static void main(String[] args) {
        Carro carroTeste = new Carro("Toyota", "Corolla", 2020);

        System.out.println(carroTeste.acelerar(50));
        System.out.println(carroTeste.frear(20));
        System.out.println(carroTeste.exibirVelocidade());
    }
}

File: Python/ex2.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def detalhes(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}"

    def acelerar(self, incremento):
        self.velocidade += incremento
        return f"{self.marca} {self.modelo} acelerou. Velocidade atual: {self.velocidade} km/h"

    def frear(self, decremento):
        self.velocidade = max(0, self.velocidade - decremento)
        return f"{self.marca} {self.modelo} freou. Velocidade atual: {self.velocidade} km/h"

    def exibir_velocidade(self):
        return f"Velocidade atual do {self.marca} {self.modelo}: {self.velocidade} km/h"


carro_teste = Carro("Toyota", "Corolla", 2020)

print(carro_teste.acelerar(50))
print(carro_teste.frear(20))
print(carro_teste.exibir_velocidade())